config/defaults.go

```diff
@@ -119,66 +119,54 @@ func (d *Defaults) Validate() error {
 // system cannot infer a run image for "fortran" if a build image was not
 // declared for this language in the Defaults object.
 func (d *Defaults) SetLoadTestDefaults(test *grpcv1.LoadTest) error {
-	var err error
-
-	spec := &test.Spec
+	testSpec := &test.Spec
+	im := newImageMap(d.Languages)
 
 	if test.Namespace == "" {
 		test.Namespace = d.ComponentNamespace
 	}
 
-	if spec.Driver == nil {
-		spec.Driver = new(grpcv1.Driver)
-	}
-
-	if spec.Driver.Language == "" {
-		// TODO: Make default driver language a configuration option
-		spec.Driver.Language = "cxx"
-	}
-
-	if spec.Driver.Run.Image == nil {
-		spec.Driver.Run.Image = &d.DriverImage
-	}
-	if err = d.setComponentDefaults(&spec.Driver.Component, d.DriverPool); err != nil {
+	if err := d.setDriverDefaults(im, testSpec.Driver); err != nil {
 		return errors.Wrap(err, "could not set defaults for driver")
 	}
 
-	for i := range spec.Servers {
-		if err = d.setComponentDefaults(&spec.Servers[i].Component, d.WorkerPool); err != nil {
+	for i := range testSpec.Servers {
+		if err := d.setServerDefaults(im, &testSpec.Servers[i]); err != nil {
 			return errors.Wrapf(err, "could not set defaults for server at index %d", i)
 		}
 	}
 
-	for i := range spec.Clients {
-		if err = d.setComponentDefaults(&spec.Clients[i].Component, d.WorkerPool); err != nil {
+	for i := range testSpec.Clients {
+		if err := d.setClientDefaults(im, &testSpec.Clients[i]); err != nil {
 			return errors.Wrapf(err, "could not set defaults for client at index %d", i)
 		}
 	}
 
 	return nil
 }
 
-// setComponentDefaults sets default name, pool and container images for a
-// component. An error is returned if a default could not be inferred for a
-// field.
-func (d *Defaults) setComponentDefaults(component *grpcv1.Component, defaultPool string) error {
-	language := component.Language
-	im := newImageMap(d.Languages)
-
-	if component.Name == nil {
-		name := uuid.New().String()
-		component.Name = &name
+// setNameOrDefault replaces the address of its pointer argument with the
+// address of a UUID string. If the argument is nil upon invocation, it will
+// remain nil.
+func (d *Defaults) setNameOrDefault(namePtr *string) {
+	if namePtr != nil {
+		return
```

Comment:
A bit confused about this function, the function itself is well self explanatory: if we namePtr is not nil we do not anything, and if it is nil we give it a UUID. This seems not consistent with the comment, could you help me understand this?

Reply:
@wanlin31 Sorry, I pushed revision 5e6d98a. Let me know if that fixes it.